compiler/damlc/tests/daml-test-files/Interface.daml

```diff
@@ -46,30 +59,18 @@ template Asset
         -- TODO https://github.com/digital-asset/daml/issues/10810
         --  (maybe) support `let setAmount x = ...` syntax.
 
-      choice Split : (ContractId Token, ContractId Token)
-        with
-          splitAmount : Int
-        controller owner
-        do
-          assert (splitAmount < amount)
-          cid1 <- create this with amount = splitAmount
-          cid2 <- create this with amount = amount - splitAmount
-          pure (toTokenContractId cid1, toTokenContractId cid2)
-
-      choice Transfer : ContractId Token
-        with
-          newOwner : Party
-        controller owner, newOwner
-        do
-          cid <- create this with owner = newOwner
-          pure (toTokenContractId cid)
-
-      nonconsuming choice Noop : ()
-        with
-          nothing : ()
-        controller owner
-        do
-          pure ()
+      let splitImpl = \splitAmount -> do
```

Comment:
So this separation of implementation and choice body is something we always have to do now, right?

Reply:
exactly! As @cocreature pointed out in [the issue](https://github.com/digital-asset/daml/issues/11372) this PR should fix, we can later add virtual choices again as syntactic sugar, but internally they would still be fixed choices with interface methods as the implementation.